src/utils/index.ts

```diff
@@ -61,23 +61,20 @@ export function constructRecords(
   return records;
 }
 
-export const domainExtensionToNamingServiceName = {
-  crypto: NamingServiceName.UNS,
-  zil: NamingServiceName.ZNS,
-};
-
 export const findNamingServiceName = (
   domain: string,
 ): NamingServiceName | '' => {
   const extension = domain.split('.').pop();
 
   if (!extension) {
     return '';
-  } else if (extension in domainExtensionToNamingServiceName) {
-    return domainExtensionToNamingServiceName[extension];
-  } else {
-    return domainExtensionToNamingServiceName.crypto;
   }
+
+  if (extension === 'zil') {
+    return NamingServiceName.ZNS;
+  }
```

Comment:
Is this going to work with zil on UNS?

Reply:
That's a good point, good catch! It was refactored to the same logic as before. We may want to pass another parameter to help with the checking 🤔